Fix iss6977: Avoid wrapping LightningModule in *DataParallel overrides when not fitting (#8632)
PyTorchLightning/pytorch-lightning: CHANGELOG.md

- Fixed an issue with `training_step` outputs not getting collected correctly for `training_epoch_end` ([#8613](https://github.com/PyTorchLightning/pytorch-lightning/pull/8613))

> - Fixed avoid wrapping LightningModule in *DataParallel overrides when not fitting ([#6977]https://github.com/PyTorchLightning/pytorch-lightning/issues/6977)

Reviewer: - Please make the changelog explicit with which plugins and data-parallel wrappers were changed to help users who search this
- Please add the changelog entry at the end of the list, along with the same formatting that other entries have

Author: I updated changelog with more details in the coming PR.

- Fixed an issue with logger outputs not being finalized correctly after prediction runs ([#8333](https://github.com/PyTorchLightning/pytorch-lightning/issues/8333))

> - Fixed avoid wrapping LightningModule in *DataParallel overrides when not fitting ([#6977]https://github.com/PyTorchLightning/pytorch-lightning/issues/6977). Specifically,
>     - Update `configure_ddp` function in `DDPPlugin`, `DDPSpawnPlugin`, `DDPShardedPlugin` and `DDPSpawnShardedPlugin` by checking the state of LightningModule and avoiding wrapping LihgningModule as *DataParallel when the state is not `TrainerFn.FITTING`.
>     - Update `validation_step` function in `DDPPlugin` and `DDPSpawnPlugin` to use LightningModule's `validation_step` function if `self.model` is not `DistributedDataParallel` instance.
>     - Update `test_step` and `prediction_step` functions in  `DDPPlugin` and `DDPSpawnPlugin` to use  LightningModule's `*_step` functions directly.
> 

Reviewer: this can be condensed down 😄 
for folks who want more context, they can check the linked PR from the changelog
```suggestion
- Fixed avoid wrapping LightningModule with data-parallel when not fitting in `DDPPlugin`, `DDPSpawnPlugin`, `DDPShardedPlugin`, `DDPSpawnShardedPlugin` ([#6977]https://github.com/PyTorchLightning/pytorch-lightning/issues/6977).
```

Author: update changelog in latest pull request.
